### ISSUE TITLE
Confusing error message when trying to extend an interface

### ISSUE THREAD

**Issue by dylanscott** [Microsoft/TypeScript#8175]
This is a somewhat cosmetic/quality of life issue, but I found the compiler error in this particular case could have been much more helpful.

**TypeScript Version:** 1.8.9

**Code**
```ts
interface A { }
class B extends A { }
```

**Expected behavior:** An error alluding to the fact that `A` cannot be extended because it is an interface.

**Actual behavior:** 
```bash
a.ts(2,17): error TS2304: Cannot find name 'A'.
```

**Comment by RyanCavanaugh** [comment]
We should do the same check we do in other places where we resolve a value-name and get nothing (so look for a namespace-name or type-name instead to give a better error)

**Comment by mhegazy** [comment]
duplicate of https://github.com/Microsoft/TypeScript/issues/7900

**Comment by mhegazy** [comment]
and https://github.com/Microsoft/TypeScript/issues/102 covers the namespace one.
